What about using WSL Rust?

Hi. I'm trying to figure out how to use rustc & cargo from my WSL, where I installed rust, and works perfectly. In fact, I don't want to install Rust on Windows.
> https://medium.com/@bgourlie/cross-platform-development-on-windows-is-suddenly-awesome-da863a28fa1e

Same issue on Arch as noted above. Here's the tail from the logs:

```
2021-05-20 22:58:48,334 [1138019]   WARN - rust.openapiext.CommandLineExt - Failed to run executable 
com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "C:\Program Files\JetBrains\IntelliJ IDEA 2021.1\jbr\bin\rustc.exe" (in directory "\\wsl$\Arch\home\jem\src"): CreateProcess error=2, The system cannot find the file specified
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:385)
	at com.intellij.execution.process.OSProcessHandler.startProcess(OSProcessHandler.java:100)
	at com.intellij.execution.process.OSProcessHandler.<init>(OSProcessHandler.java:48)
	at com.intellij.execution.process.CapturingProcessHandler.<init>(CapturingProcessHandler.java:24)
	at org.rust.cargo.runconfig.RsCapturingProcessHandler.<init>(RsCapturingProcessHandler.kt:12)
	at org.rust.openapiext.CommandLineExtKt.execute(CommandLineExt.kt:32)
	at org.rust.cargo.toolchain.tools.Rustc.getSysroot(Rustc.kt:44)
	at org.rust.cargo.toolchain.tools.Rustc.getStdlibPathFromSysroot(Rustc.kt:51)
	at org.rust.cargo.toolchain.tools.Rustc.getStdlibFromSysroot(Rustc.kt:56)
	at org.rust.cargo.project.settings.ui.RustProjectSettingsPanel$update$1.invoke(RustProjectSettingsPanel.kt:121)
	at org.rust.cargo.project.settings.ui.RustProjectSettingsPanel$update$1.invoke(RustProjectSettingsPanel.kt:34)
	at org.rust.openapiext.UiDebouncer$run$1.run(ui.kt:42)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:369)
	at com.intellij.util.Alarm$Request.run(Alarm.java:355)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Cannot run program "C:\Program Files\JetBrains\IntelliJ IDEA 2021.1\jbr\bin\rustc.exe" (in directory "\\wsl$\Arch\home\jem\src"): CreateProcess error=2, The system cannot find the file specified
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at com.intellij.execution.configurations.GeneralCommandLine.startProcess(GeneralCommandLine.java:458)
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:381)
	... 29 more
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.base/java.lang.ProcessImpl.create(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:478)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:154)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 32 more
2021-05-20 22:58:51,277 [1140962]   INFO - s.impl.wsl.WslFileWatcher.Arch - Watcher terminated with exit code 0 
2021-05-20 22:58:53,228 [1142913]   INFO - rust.openapiext.CommandLineExt - Executing `C:\WINDOWS\system32\wsl.exe --distribution Arch --exec /bin/sh -c "export PATH=\"/usr/bin:$PATH\" && /usr/bin/rustc --version --verbose"` 
2021-05-20 22:58:53,466 [1143151]   INFO - rust.openapiext.CommandLineExt - Executing `C:\WINDOWS\system32\wsl.exe --distribution Arch --exec /bin/sh -c "export PATH=\"/usr/bin:$PATH\" && cd /home/jem/src && /usr/bin/rustc --print sysroot"` 
2021-05-20 22:59:03,989 [1153674]   WARN - options.ex.ConfigurableWrapper - Cannot find extension point com.intellij.compilerSettingsFactory in Project(name=src, containerState=COMPONENT_CREATED, componentStore=\\wsl$\Arch\home\jem\src) 
```

--- a/src/main/resources/META-INF/rust-core.xml
+++ b/src/main/resources/META-INF/rust-core.xml
@@ -1137,7 +1137,7 @@
         <console.folding implementation="org.rust.ide.console.RsConsoleFolding"/>
 
         <!--  Experimental features -->
-        <experimentalFeature id="org.rust.wsl" percentOfUsers="0">
+        <experimentalFeature id="org.rust.wsl" percentOfUsers="100">
             <description>Enables WSL toolchains</description>
         </experimentalFeature>
         <experimentalFeature id="org.rust.cargo.features.settings.gutter" percentOfUsers="0">

